pkg/cable/wireguard/README.md

```diff
@@ -0,0 +1,53 @@
+# WireGuard Cable Driver (WIP)
+
+[WireGuard](https://www.wireguard.com "WireGuard homepage") is an extremely simple yet fast and modern VPN that utilizes state-of-the-art cryptography.
+
+Traffic is encrypted and encapsulated in UDP packets.
+
+## Driver design
+
+- WireGuard creates a virtual network device that is accessed via netlink. It appears like any network device and currently has a hardcoded name `subwg0`.
+
+- WireGuard identifies peers by their cryptographic public key without the need to exchange shared secrets. The owner of the public key must have the corresponding private key to prove identity.
+
+- The driver creates the key pair and adds the public key to the local endpoint so other clusters can connect. Like `ipsec`, the node IP address is used as the endpoint udp address of the WireGuard tunnels. A fixed port is used for all endpoints.
+
+- The driver adds routing rules to redirect cross cluster communication through the virtual network device `subwg0`.
+  (*note: this is different from `ipsec`, which intercepts packets at netfilter level.*)
+
+- The driver uses [`wgctrl`](https://github.com/WireGuard/wgctrl-go "WgCtrl github"), a go package that enables control of WireGuard devices on multiple platforms. Link creation and removal are done through [`netlink`](https://github.com/vishvananda/netlink "Netlink github").
+
+## Installation
+
+- WireGuard needs to be [installed](https://www.wireguard.com/install "WireGuard installation instructions") on the gateway nodes. For example, (Ubuntu < 19.04),
+  ```ShellSession
+  $ sudo add-apt-repository ppa:wireguard/wireguard
+  $ sudo apt-get update
+  $ sudo apt-get install wireguard
+  ```
+
+- The driver needs to be enabled with
+  ```ShellSession
+  $ bin/subctl join --kubeconfig wg3-conf --cable-driver wireguard --disable-nat  --repository roytman --version  latest broker-info.subm
```

Comment:
I'd drop the repo & version flags (unless they have some significance specifically for WG?)

Reply:
They are no longer significant, as latest subctl now supports the --cable-driver option. Removed.